### DISCUSSION TITLE
File Extraction

### DISCUSSION THREAD

**Discussion by lovejotd** [chmorgan/sharppcap#326] [Q&A] [answered]
I was wondering is it possible to use sharppcap to extract files from pcap?  For example, images or other files via ftp.  Thanks!

**Comment by kayoub5** [comment, marked as answer]
@chmorgan I think this question should be addressed/moved to https://github.com/chmorgan/packetnet-connections

**Comment by chmorgan** [comment]
@kayoub5 @lovejotd yep. packetnet-connections can follow tcp connections and decode http. I think there are some examples showing how to get web pages downloaded or something in there. You'd have to add a protocol dissector for ftp transfers though.

**Reply by lovejotd** [reply]
Do you have any examples of this? Or anything?  Also, does it work with pcap files?  Thanks!

**Comment by chmorgan** [comment]
https://github.com/chmorgan/packetnet-connections has some screen
recordings of the examples and other info, examples are in the examples
folder.

**Reply by lovejotd** [reply]
So I dont see in any of the examples actually getting the file from the data received.  When I try to do this, the e.request.body in my handlers is always null (never any data), although much of the other data is filed in.  Should this be the case?  Is this something I need to track myself?  Thanks!
